expressions.go

```diff
@@ -319,6 +319,36 @@ var functions = map[string]struct {
 			return nil, fmt.Errorf("sum expects argument #1 to be slice of numbers")
 		}
 	}},
+	"sumiftrue": {2, func(args []Value) (Value, error) {
+		arg := args[0]
+		condition := args[1]
+		switch v := arg.(type) {
+		case *Slice:
+			numType, ok := v.typ.elementType.(*NumberType)
+			if !ok {
+				return nil, fmt.Errorf("sumiftrue expects argument #1 to be slice of numbers")
+			}
+			conditionSlice := condition.(*Slice)
+			_, ok = conditionSlice.typ.elementType.(*BoolType)
+			if !ok {
+				return nil, fmt.Errorf("sumiftrue expects argument #2 to be slice of bools")
+			}
+			sum := &Number{0, numType}
+			for i, elem := range v.elements {
```

Comment:
nit: since we're iterating over two slices, the explicit `for i := 0; i < sliceslen; i++ {` would be preferred